frontend/src/features/SideWindow/Alert/SilencedAlerts/schemas.ts

```diff
@@ -0,0 +1,64 @@
+import { logSoftError } from '@mtes-mct/monitor-ui'
+import { without } from 'lodash'
+import { addMethod, number, object, string } from 'yup'
+
+import type { SilencedAlertData } from '../../../../domain/entities/alerts/types'
+
+/**
+ * Require at least one of the specified fields to be required
```

Comment:
Cette partie est assez complexe.

- Je typerais le contexte : `atLeastOneRequired(this: this: ObjectSchema<{ [key: string]: any }>, list: string[], message: string)`
- Je nommerais le param `list` => `requiredFields`
- Je splitterais le code en deux étapes pour le rendre plus clair (en sortant les fonctions utilisées dans les reducers pour les décrire et les typer, ce qui évitera peut-être en bonus les @ts-ignore)
- Description => "Require at least one of the specified fields to be present."
- Je ne suis pas sûr de bien comprendre (l'utilité de) `list.reduce((acc, item, idx, all) => [...acc, ...all.slice(idx + 1).map(i => [item, i])], [])`
- Ca mérite une petit exemple explicatif et un test unitaire ^^

Reply:
Disclaimer: je n'ai pas écris ce code ^^
J'ai rajouté un test pour valider son fonctionnement mais je ne comprends pas (comme toi) la dernière ligne, donc je propose de le laisser en mode "black box", ce code devrait clairement faire partie d'une lib, et la logique de Yup est assez complexe je trouve.

Reply:
Je te propose alors de simplement mettre à jour la description et de mettre un todo+comment pour qu'éventuellement on voit ça plus tard si ça te va ^^. Je suis d'accord que ça deviendra une sorte de lib ou plutôt un util dans monitor-ui.